tests/test-analyser.el

```diff
@@ -29,7 +29,7 @@
 
       (it "should analyze the default form of the defconst and assign that type"
         (elsa-test-with-analysed-form "(progn (defconst foo 'bar) foo)" form
-          (expect (elsa-nth 2 form) :to-be-type-equivalent (elsa-make-type Const bar))))
+          (expect (elsa-get-type (elsa-nth 2 form)) :to-equal (elsa-make-type Readonly Const bar))))
```

Comment:
Shouldn't this also use the type equivalency check?  I'm not sure if equality will match in the new Emacs versions with the structures in place of vectors.

But I think the problem is that readonly accepts nothing and so we can't really use this helper... hmm.

Maybe test that it's readonly with the predicate and then test the wrapped types?

Reply:
> Maybe test that it's readonly with the predicate and then test the wrapped types?

Sounds reasonable.  In the long run I think it's useful to have a strict equality as well, especially for tests.